### PR TITLE
Use the grpc internal backoff mechanisms for connection retry

### LINKED ISSUE #259: Use GRPC's supported connection options to drive retries on failure
Remove the manual backoff logic and use the GRPC included options.

<sup>Created from JetBrains using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=issue&utm_campaign=github)</sup>

### PR DESCRIPTION
**Pull request by Jim3Things**
Also fixed two small issues that I found when testing stop/restart of sim_supportd while the UI was active.

closes #259

### CODE REVIEW

File: simulation/internal/services/frontend/frontend.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/Jim3Things/CloudChamber/simulation/internal/clients/limits"
```
Comment (manpreet85) [inline simulation/internal/services/frontend/frontend.go:28]: Is it in the correct place?

Comment (Jim3Things) [reply simulation/internal/services/frontend/frontend.go]: Nope, it's out of order. I've fixed this and the other places it happened.

File: simulation/internal/services/frontend/logs.go [hunk 2]
```diff
@@ -60,7 +60,7 @@ func handlerLogsGetAfter(w http.ResponseWriter, r *http.Request) {
 
 	data := <-ch
 	if data.Err != nil {
-		postHTTPError(ctx, w, err)
+		postHTTPError(ctx, w, data.Err)
 		return
 	}
```
Comment (MikeCJohnsonUk) [inline simulation/internal/services/frontend/logs.go:65]: Can the channel have been closed *before* anything was written to it? That is, can this read from the channel fail, i.e. is "data" garunteed non-nil?

Comment (Jim3Things) [reply simulation/internal/services/frontend/logs.go:65]: Yes, the channel should have one message in it before it is closed.  I didn't check for nil under the theory that if the internal consistency fails in this way, I'd rather have a panic...

File: simulation/internal/services/stepper/stepper_test.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ import (
 	"testing"
 	"time"
 
+	"github.com/Jim3Things/CloudChamber/simulation/internal/clients/limits"
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply simulation/internal/services/stepper/stepper_test.go]: This should have already been done - old version?

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**